docs/RegistryCoordinator.md

```diff
@@ -0,0 +1,325 @@
+[core-dmgr-docs]: https://github.com/Layr-Labs/eigenlayer-contracts/blob/m2-mainnet/docs/core/DelegationManager.md
+[core-dmgr-register]: https://github.com/Layr-Labs/eigenlayer-contracts/blob/m2-mainnet/docs/core/DelegationManager.md#registeroperatortoavs
+[core-dmgr-deregister]: https://github.com/Layr-Labs/eigenlayer-contracts/blob/m2-mainnet/docs/core/DelegationManager.md#deregisteroperatorfromavs
+
+## RegistryCoordinator
+
+| File | Type | Proxy? |
+| -------- | -------- | -------- |
+| [`RegistryCoordinator.sol`](../src/RegistryCoordinator.sol) | Singleton | Transparent proxy |
+
+The `RegistryCoordinator` has three primary functions:
+1. It is the primary entry and exit point for operators as they register for and deregister from quorums, and manages registration and deregistration in the `BLSApkRegistry`, `StakeRegistry`, and `IndexRegistry`. It also hooks into the EigenLayer core contracts, updating the core `DelegationManager` when an Operator registers/deregisters.
+2. It allows anyone to update the current stake of any registered operator
+3. It allows the Owner to initialize and configure new quorums
+
+#### High-level Concepts
+
+This document organizes methods according to the following themes (click each to be taken to the relevant section):
+* [Registering and Deregistering](#registering-and-deregistering)
+* [Updating Registered Operators](#updating-registered-operators)
+* [System Configuration](#system-configuration)
+
+#### Roles
+
+* Owner: a permissioned role that can create and configure quorums as well as manage other roles
+* Ejector: a permissioned role that can forcibly eject an operator from a quorum via `RegistryCoordinator.ejectOperator`
+* Churn Approver: a permissioned role that signs off on operation churn in `RegistryCoordinator.registerOperatorWithChurn`
+
+---    
+
+### Registering and Deregistering
+
+These methods allow operators to register for/deregister from one or more quorums, and are the primary entry points to the middleware contracts as a whole:
+* [`registerOperator`](#registeroperator)
+* [`registerOperatorWithChurn`](#registeroperatorwithchurn)
+* [`deregisterOperator`](#deregisteroperator)
+* [`ejectOperator`](#ejectoperator)
+
+#### `registerOperator`
+
+```solidity
+function registerOperator(
+    bytes calldata quorumNumbers,
+    string calldata socket,
+    SignatureWithSaltAndExpiry memory operatorSignature
+) 
+    external 
+    onlyWhenNotPaused(PAUSED_REGISTER_OPERATOR)
+```
```

Comment:
unfortunately my PR related to the compendium changes these functions a bit 😢  I can make a follow-up PR after these are both merged though, to make the docs more up-to-date

Reply:
all good, I was already planning to spend time over xmas to get all the docs up to date

Reply:
esp since we just keep opening PRs :grimacing: